MapboxNavigation/RouteMapViewController.swift

```diff
@@ -409,3 +422,9 @@ extension RouteMapViewController: RoutePageViewControllerDelegate {
         return routeController.routeProgress.currentLegProgress.stepAfter(step)
     }
 }
+
+protocol RouteMapViewControllerDelegate: class {
+    func navigationMapView(_ mapView: NavigationMapView, routeStyleLayerWithIdenitier identifier: String, source: MGLSource) -> MGLStyleLayer?
```

Comment:
Normally you’d expect the first argument to be the RouteMapViewController itself: `routeMapViewController(_:navigationMapView:routeStyleLayerWithIdentifier:source:)`. That would let the developer vary the behavior from route map view controller to route map view controller. But that would be really unwieldy, and it’s relatively unlikely that the developer would be dealing with more than one route view controller at a time.